### PR TITLE
theming for form components

### PR DESCRIPTION
**Pull request by jonborchardt**
mostly snapshot changes, ignore the snapshots...

### CODE REVIEW

File: components/auto-complete/__tests__/__snapshots__/demo.test.js.snap [hunk 1]
```diff
@@ -271,24 +271,13 @@ exports[`renders ./components/auto-complete/demo/form-debug.md correctly 1`] = `
...
-                  height="1em"
-                  viewBox="64 64 896 896"
-                  width="1em"
-                >
-                  <path
-                    d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
-                  />
-                </svg>
+                <img
+                  alt=""
```
Comment (codeviking) [inline components/auto-complete/__tests__/__snapshots__/demo.test.js.snap:278]: I wish I could tell Git to just ignore snapshots at PR time.

File: components/icons/style/index.less [hunk 2]
```diff
@@ -1,2 +1,11 @@
 // placeholder
 @import '../../style/themes/index';
+
+.no-pointer-events {
+    &&&, img {
+        pointer-events: none;
+    }
+    img {
+        vertical-align: unset;
+    }
```
Comment (codeviking) [inline components/icons/style/index.less:10]: This feels like we're reversing a parent style, if so, I'd propose that the selector include the style that causes whatever we're reversing, i.e.

```
.no-pointer-events {
   &&&, img { pointer-events: none }
}
.some-parent-class {
   .no-pointer-events {
       img {
          vertical-align: unset;
       }
   }
}
```

It's more verbose, but I think clarifies the intent (though I could be wrong on the intent).

Comment (jonborchardt) [reply components/icons/style/index.less:10]: the icon can be used anywhere...

Comment (codeviking) [reply components/icons/style/index.less:10]: Yea, I just don't understand what we're `unset`ing here.

I imagine there's CSS being applied by a parent class that we're reversing -- or maybe we're just adding this in-case `vertical-align` is set somewhere in the parent, and if so why not just override it when necessary?

Comment (jonborchardt) [reply components/icons/style/index.less:10]: i take your point and will look into it

**Pull request merged**